timesketch/api/v1/resources/data.py

```diff
@@ -0,0 +1,229 @@
+from flask import abort
+from flask_restful import Resource
+from flask_login import login_required
+from flask_login import current_user
+
+from timesketch.api.v1 import resources
+from timesketch.lib.definitions import HTTP_STATUS_CODE_CREATED
+from timesketch.lib.definitions import HTTP_STATUS_CODE_BAD_REQUEST
+from timesketch.lib.definitions import HTTP_STATUS_CODE_FORBIDDEN
+from timesketch.lib.definitions import HTTP_STATUS_CODE_NOT_FOUND
+from timesketch.models import db_session
+from timesketch.models.sketch import Sketch
+from timesketch.models.sketch import Timeline
+from timesketch.models.sketch import DataSource
+
+
+logger = logging.getLogger('timesketch.datasource_api')
+
+
+class DataSourceListResource(resources.ResourceMixin, Resource):
+    """Resource for listing DataSources associated with a Sketch."""
+
+    @login_required
+    def get(self, sketch_id):
+        """Handles GET request to the resource.
+
+        Args:
+          sketch_id (int): Identifier for the Sketch the datasource belongs to.
+
+        Returns:
+            A list of JSON representations of the data sources.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                'No sketch found with this ID.')
+
+        if sketch.get_status.status == 'archived':
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Unable to fetch data sources from an archived sketch.')
+
+        number_of_timelines = 0
+        data_sources = []
+        for timeline in sketch.active_timelines:
+            number_of_timelines += 1
+            for data_source in timeline.datasources:
+                data_sources.append(data_source)
+
+        schema = {
+            'meta': {
+                'number_of_timelines': number_of_timelines,
+                'number_of_sources': len(data_sources)
+            },
+            'objects': data_sources,
+        }
+        return jsonify(schema)
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Args:
+          sketch_id (int): Identifier for the Sketch the datasource belongs to.
+
+        Returns:
+            A datasource in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                'No sketch found with this ID.')
+
+        if sketch.get_status.status == 'archived':
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Unable to fetch data sources from an archived sketch.')
+
+        if not sketch.has_permission(current_user, 'write'):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, (
+                    'User does not have sufficient access rights to '
+                    'delete a sketch.'))
```

Comment:
Wrong string here. We are not deleting a sketch :)

Reply:
no, we are not...